### ISSUE TITLE
Move folder only moves files in folder

### ISSUE THREAD

**Issue by xibriz** [vgrem/office365-rest-python-client#313]
https://github.com/vgrem/Office365-REST-Python-Client/blob/4999e197c64d3f8f7fbcb7650a9e10b949dbf632/office365/sharepoint/folders/folder.py#L123-L139

File structure:

```
MySite
|-> Folder1
|--> File1
|---> Subfolder1
|----> File2
|-> Folder2
```

Code:

```
source_url = '/Shared Documents/Folder1'
destination_url = '/sites/MySite/Shared Documents/Folder2'

folder = contex.web.get_folder_by_server_relative_url(source_url)
folder.move_to(destination_url , 1)
contex.execute_query()
```

This moves `File1` into `Folder2`, but I expect to find `Folder1` inside `Folder2`.

If I change `source_url` to `/sites/MySite/Shared Documents/Folder1` nothing happens.
If I change `destination_url` to `/Shared Documents/Folder2` nothing happens.
If I change `destination_url` to `/sites/MySite/Shared Documents/Folder2/Folder1` I get `'Folder "Folder1" does not exist.'`

**Comment by vgrem** [comment]
Greetings!

This is rather an expected behavior (not a bug definitely) since `Folder.move_to method` expects _existing target folder_ otherwise the provided error occurs. For the scenario, i propose to call method to ensure a target folder exist:

> `Web.ensure_folder_path(path) `- Ensures a nested folder hierarchy exist



_Updated_ example:

```
source_url = '/Shared Documents/Folder1'
destination_url = '/Shared Documents/Folder2'
context.web.ensure_folder_path(destination_url).execute_query()  

folder = contex.web.get_folder_by_server_relative_url(source_url)
folder.move_to(destination_url , 1)
contex.execute_query()
```




But if your folder needs to be moved **across site collections**,  `Folder.move_to` is no use here,   the following example demonstrates how to accomplish it:

```
source_ctx = ClientContext("/sites/site_from").with_credentials(credentials)

source_folder_url = "/Shared Documents"   #resides in site /sites/site_from
target_folder_url = "/Shared Documents"   #resides in site /sites/site_to

source_folder = source_ctx.web.get_folder_by_server_relative_url(source_folder_url).expand(["Files"]).get().execute_query()

target_ctx = ClientContext("/sites/site_to").with_credentials(credentials)
target_folder = target_ctx.web.ensure_folder_path(target_folder_url)

with tempfile.TemporaryDirectory() as local_path:
    for file in source_folder.files:  # type: File
        print(f"Copying file from: {file.serverRelativeUrl}")
        file_name = os.path.basename(file.serverRelativeUrl)
        file_path = os.path.join(local_path, file_name)
        with open(file_path, 'wb') as f:
            file_content = file.download(f).execute_query()
        target_file = target_folder.upload_file(file_name, file_content).execute_query()
        print(f"File copied into {target_file.serverRelativeUrl}")

```

**Comment by xibriz** [comment]
@vgrem can't say I agree that this is resolved. I am trying to move a folder with all sub files and sub folders to a new location within the same site.

Your updated example only moved the files inside `Folder1` directly into `Folder2`, leaving `Folder1` at the same source location with no files.

And if `Folder1` has any sub folders with or without files, these are also not moved.

**Comment by xibriz** [comment]
Looking at the code referenced in the first comment it is clear that this library don't even try to move any folders as its just looping over files in the source folder and moves them individually.

I'm guessing that is because the folder object don't have any move methods
https://docs.microsoft.com/en-us/previous-versions/office/sharepoint-visio/jj245697(v=office.15)?redirectedfrom=MSDN

If there are not any APIs to move a folder this library should (in my opinion) loop trough the source_folder and ensure the same folders at the destination folder, then move the files folder by folder.

**Comment by vgrem** [comment]
You are right, `Folder.move_to `supports moving files within folder _1 level beneath only_.

For the case of moving  the whole folder hierarchy, `MoveCopyUtil.copyFolder` needs to be utilized instead ([here are provided some details](https://collab365.community/sp2016-o365-sp-movecopyutil-javascript-copy-move-files-and-folder/]), will provide an example later today.

**Comment by vgrem** [comment]
Not a library bug, this is SharePoint REST API behaviour. `MoveTo` destination URL semantics:

- Destination = `/Folder2` → moves **contents of Folder1 into Folder2** (files, not the folder itself)
- Destination = `/Folder2/Folder1` → moves **Folder1 itself into Folder2**

The confusion is that SharePoint's `MoveTo` destination is the **final path of the moved object**, not the parent folder.

**Correct pattern - use `move_to_using_path` (recommended in v3.0.0):**

```python
from office365.sharepoint.client_context import ClientContext

ctx = ClientContext(site_url).with_client_certificate(
    tenant, client_id, thumbprint, cert_path
)

# move Folder1 into Folder2 - destination includes the folder name
ctx.web.get_folder_by_server_relative_url(
    "/sites/MySite/Shared Documents/Folder1"
).move_to_using_path(
    "/sites/MySite/Shared Documents/Folder2/Folder1"  # ← full destination path
).execute_query()
```

Result:
```
MySite
|-> Folder2
|---> Folder1       ← moved here, including Subfolder1 and all contents
|-----> File1
|-----> Subfolder1
|-------> File2
```

`move_to_using_path` uses `DecodedUrl` instead of `serverRelativeUrl` - handles spaces and special characters correctly and is the preferred method over the older `moveto`.
